### ISSUE TITLE
Validate icons attribute

### ISSUE THREAD

**Issue by pdehaan** [mozilla/addons-linter#846]
I'm a bit confused by the `icons` entry in a manifest.json file.

- W3C Spec: https://www.w3.org/TR/appmanifest/#icons-member

    > The icons member is an array of image objects that can serve as iconic representations of the web application in various contexts.

- MDN Docs: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/icons

    > It consists of key-value pairs of image size in px and image path relative to the root directory of the extension.

- Chrome Docs:https://developer.chrome.com/extensions/manifest/icons

    ```js
"icons": {
      "16": "icon16.png",
      "48": "icon48.png",
      "128": "icon128.png"
},
```

- Online validator: https://manifest-validator.appspot.com/

  > ERROR: 'icons' expected to be an array but is not.

The docs and specs are a bit vague as to whether the `icons` attribute should be an Object or an Array.

**Comment by andymckay** [comment]
The MDN and Chrome docs seem pretty much in sync:

https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/icons

You are looking at the app spec and an app validator, not the extension one. The W3C proposal (still very early days) is here: 

https://lists.w3.org/Archives/Public/public-browserext/2016May/0000.html

And proposes an object as well.
